internal/api/deployments.go

```diff
@@ -0,0 +1,110 @@
+	List(ctx context.Context, handleNames []string) ([]*Deployment, error)
+	Update(ctx context.Context, deploymentID uuid.UUID, data DeploymentUpdate) error
+	Delete(ctx context.Context, deploymentID uuid.UUID) error
+}
+
+// Deployment is a representation of a deployment.
+type Deployment struct {
+	BaseModel
+	AccountID   uuid.UUID `json:"account_id"`
+	WorkspaceID uuid.UUID `json:"workspace_id"`
+
+	Description            string    `json:"description,omitempty"`
+	EnforceParameterSchema bool      `json:"enforce_parameter_schema"`
+	Entrypoint             string    `json:"entrypoint"`
+	FlowID                 uuid.UUID `json:"flow_id"`
+	ManifestPath           string    `json:"manifest_path,omitempty"`
+	Name                   string    `json:"name"`
+	Path                   string    `json:"path"`
+	Paused                 bool      `json:"paused"`
+	Tags                   []string  `json:"tags"`
+	Version                string    `json:"version,omitempty"`
+	WorkPoolName           string    `json:"work_pool_name,omitempty"`
+	WorkQueueName          string    `json:"work_queue_name,omitempty"`
+}
+
+// DeploymentCreate is a subset of Deployment used when creating deployments.
+type DeploymentCreate struct {
+	Description            string    `json:"description,omitempty"`
+	EnforceParameterSchema bool      `json:"enforce_parameter_schema,omitempty"`
+	Entrypoint             string    `json:"entrypoint,omitempty"`
+	FlowID                 uuid.UUID `json:"flow_id"`
+	ManifestPath           string    `json:"manifest_path,omitempty"`
+	Name                   string    `json:"name"`
+	Path                   string    `json:"path,omitempty"`
+	Paused                 bool      `json:"paused,omitempty"`
+	Tags                   []string  `json:"tags,omitempty"`
+	Version                string    `json:"version,omitempty"`
+	WorkPoolName           string    `json:"work_pool_name,omitempty"`
+	WorkQueueName          string    `json:"work_queue_name,omitempty"`
+}
+
+// DeploymentUpdate is a subset of Deployment used when updating deployments.
+type DeploymentUpdate struct {
+	Description            string   `json:"description,omitempty"`
+	EnforceParameterSchema bool     `json:"enforce_parameter_schema,omitempty"`
+	Entrypoint             string   `json:"entrypoint,omitempty"`
+	ManifestPath           string   `json:"manifest_path"`
+	Path                   string   `json:"path,omitempty"`
+	Paused                 bool     `json:"paused,omitempty"`
+	Tags                   []string `json:"tags,omitempty"`
+	Version                string   `json:"version,omitempty"`
+	WorkPoolName           string   `json:"work_pool_name,omitempty"`
+	WorkQueueName          string   `json:"work_queue_name,omitempty"`
+}
+
+// DeploymentFilter defines the search filter payload
+// when searching for deployements by name.
+// example request payload:
+// {"deployments": {"handle": {"any_": ["test"]}}}.
+type DeploymentFilter struct {
+}
```

Comment:
There are some extra structs here and below that aren't implemented yet, but I didn't want to delete them and have to recreate them later. Happy to do that though if desired.

Reply:
happy to keep them in